src/utils/metrics.cpp

```diff
@@ -177,6 +203,109 @@ void metric_entity::take_snapshot(metric_json_writer &writer, const metric_filte
     writer.EndObject();
 }
 
+bool metric_entity::is_stale() const
+{
+    // Since this entity itself is still being accessed, its reference count should be 1
+    // at least.
+    CHECK_GE(get_count(), 1);
+
+    // Once this entity did not have any metric, and had only one reference kept in the
+    // registry, this entity would be considered useless.
+    return _metrics.empty() && get_count() == 1;
+}
+
+metric_entity::collected_old_metrics_info metric_entity::collect_old_metrics() const
```

Comment:
what dose 'old' mean?

Reply:
`old` means a metric or entity is no longer used. Typically a metric or entity is referenced by a class. For example, the class `replica` may use a metric as the measurement of QPS for a replica.

However, once the replica is removed, this metric will not be used. The framework of metrics then will find this metric and mark it `old`. After a configurable retention interval, this metric will finally be removed from the repository of metrics.